pkg/runtime/service_controller.go

```diff
@@ -87,6 +92,44 @@ func (c *serviceController) GetResourceManagerFactories() map[string]acktypes.AW
 	return c.rmFactories
 }
 
+// GetAdoptedResourceInstalled returns whether the AdoptedResource CRD has been
+// installed into the cluster, and is accessible by the service controller.
+func (c *serviceController) GetAdoptedResourceInstalled(mgr ctrlrt.Manager) (bool, error) {
+	clusterConfig := mgr.GetConfig()
+	clientSet, err := kubernetes.NewForConfig(clusterConfig)
+	if err != nil {
+		return false, err
+	}
+
+	gv := schema.GroupVersion{
+		Group:   ackv1alpha1.GroupVersion.Group,
+		Version: ackv1alpha1.GroupVersion.Version,
+	}
+
+	// Ensure GV is supported
+	if err = discovery.ServerSupportsVersion(clientSet, gv); err != nil {
+		return false, nil
```

Comment:
Maybe i'm missing something but i think this should be a return `return false, err`

Reply:
I think for where this method is called, some errors are blocking and some are non-blocking, based on the `if/else` logic where this is called. Maybe, inline comments in this method are needed to avoid confusion and state which error blocks and which does now.